File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t){
        int s_new = 0;
        int t_new = 0;
        while(s_new < s.length() && t_new < t.length()){
            if(s.charAt(s_new) == t.charAt(t_new)){
                s_new++;
            }
            t_new++;
        }
        if(s_new == s.length()){
            return true;
        }
        return false;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        indexI = 0
        indexJ = 0
        while indexI < len(s) and indexJ < len(t):
            if s[indexI] == t[indexJ]:
                indexI += 1
            indexJ += 1
        return indexI == len(s)


